azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -377,69 +381,97 @@ def __get_agent_cgroups(self, agent_slice, cpu_controller_root, memory_controlle
+            if CGroupConfigurator._Impl.__try_set_cpu_quota(''):  # setting an empty value resets to the default (infinity)
+                CGroupsTelemetry.set_track_throttled_time(False)
 
-        def _set_cpu_quota(self, quota):
+        @staticmethod
+        def __try_set_cpu_quota(quota):
             try:
                 drop_in_file = os.path.join(systemd.get_agent_drop_in_path(), _AGENT_DROP_IN_FILE_CPU_QUOTA)
                 contents = _AGENT_DROP_IN_FILE_CPU_QUOTA_CONTENTS_FORMAT.format(quota)
-                self._create_unit_file(drop_in_file, contents)
+                CGroupConfigurator._Impl.__create_unit_file(drop_in_file, contents)
             except Exception as exception:
                 logger.info('Failed to set CPUQuota: {0}', ustr(exception))
                 return False
             try:
                 logger.info("Executing systemctl daemon-reload...")
                 shellutil.run_command(["systemctl", "daemon-reload"])
             except Exception as exception:
-                self.__log_cgroup_warning("daemon-reload failed (set quota): {0}", ustr(exception))
+                _log_cgroup_warning("daemon-reload failed (set quota): {0}", ustr(exception))
                 return False
             return True
 
-        def check_processes_in_agent_cgroup(self):
+        def check_cgroups(self, cgroup_metrics):
+            if not self.enabled():
+                return
+
+            errors = []
+
+            try:
+                self._check_processes_in_agent_cgroup()
+            except CGroupsException as exception:
+                errors.append(exception)
+
+            try:
+                self._check_agent_throttled_time(cgroup_metrics)
+            except CGroupsException as exception:
+                errors.append(exception)
+
+            if len(errors) > 0:
+                self.disable("Check on cgroups failed:\n{0}".format("\n".join([ustr(e) for e in errors])))
+
+        def _check_processes_in_agent_cgroup(self):
             """
             Verifies that the agent's cgroup includes only the current process, its parent, commands started using shellutil and instances of systemd-run
             (those processes correspond, respectively, to the extension handler, the daemon, commands started by the extension handler, and the systemd-run
             commands used to start extensions on their own cgroup).
             Other processes started by the agent (e.g. extensions) and processes not started by the agent (e.g. services installed by extensions) are reported
             as unexpected, since they should belong to their own cgroup.
-            """
-            if not self.enabled():
-                return True
 
-            def log_message(message):
-                # Report only a small sample of errors
-                if message != self._check_processes_in_agent_cgroup_last_error and self._check_processes_in_agent_cgroup_error_count < 5:
```

Comment:
We were keeping track of previous errors to avoid being too verbose when logging/sending telemetry. Since now we disable cgroups when this check fails, there is no need to keep track of them.